cell_based/src/population/ImmersedBoundaryCellPopulation.hpp

```diff
@@ -0,0 +1,679 @@
+     */
+    virtual void WriteVtkResultsToFile(const std::string& rDirectory);
+
+    friend class boost::serialization::access;
+    /**
+     * Serialize the object and its member variables.
+     *
+     * Note that serialization of the mesh and cells is handled by 
+     * load/save_construct_data.
+     *
+     * Note also that member data related to writers is not saved - output must
+     * be set up again by the caller after a restart.
+     *
+     * @param archive the archive
+     * @param version the current version of this class
+     */
+    template <class Archive>
+    void serialize(Archive& archive, const unsigned int version)
+    {
+        archive& boost::serialization::base_object<AbstractOffLatticeCellPopulation<DIM> >(*this);
+        archive& mpImmersedBoundaryDivisionRule;
+        archive& mPopulationHasActiveSources;
+        archive& mInteractionDistance;
+        archive& mReMeshFrequency;
+        archive& mThrowStepSizeException;
+        archive& mCellRearrangementThreshold;
+    }
+
+    /**
+     * Helper method. Calculate the discrete delta approximation based on 
+     * distance and grid spacing.
+     * 
+     * @param dist distance between the two points
+     * @param spacing the grid spacing
+     * @return computed Delta1D value
+     */
+    double Delta1D(double dist, double spacing);
+
+    /**
+     * Helper method for the constructor. Calculate an intrinsic length scale 
+     * based on the size of elements in the mesh.
+     * @return computed intrinsic length scale
+     */
+    double CalculateIntrinsicCellSize();
+
+    /**
+     * Check the consistency of internal data structures. Each 
+     * ImmersedBoundaryElement must have a CellPtr associated with it.
+     */
+    void Validate();
+
+public:
+    /**
+     * Create a new cell population facade from a mesh and collection of cells.
+     *
+     * There must be precisely one CellPtr for each ImmersedBoundaryElement in
+     * the mesh.
+     *
+     * @param rMesh reference to an immersed boundary mesh. Each element in the mesh
+     * will correspond to a cell.
+     * @param rCells reference to a vector of CellPtrs
+     * @param deleteMesh set to true if you want the cell population to free the 
+     *     mesh memory on destruction (defaults to false)
+     * @param validate whether to validate the cell population when it is 
+     *     created (defaults to true)
+     * @param locationIndices an optional vector of location indices that 
+     *     correspond to real cells
+     */
+    ImmersedBoundaryCellPopulation(ImmersedBoundaryMesh<DIM, DIM>& rMesh,
+                                   std::vector<CellPtr>& rCells,
+                                   bool deleteMesh = false,
+                                   bool validate = true,
+                                   const std::vector<unsigned> locationIndices = std::vector<unsigned>());
+
+    /**
+     * Constructor for use by boost serialization ONLY!
```

Comment:
If it is boost serialization only then should it be private, I think friends can access private methods?

Reply:
This being public is in keeping with existing equivalents in other files, e.g. https://github.com/Chaste/Chaste/blob/babb7e0881f31c6652dd82214666d0c40c5026b9/cell_based/src/population/MeshBasedCellPopulation.hpp#L184-L190